### PR TITLE
feat: [CDS-40523]: Added user fields for jira steps

### PR DESCRIPTION
**Pull request by sonaligoyal-harness**
#### Screenshots

https://user-images.githubusercontent.com/98508399/187901955-0233e9d7-d42b-4f0d-a907-0f1de7b74ba7.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 126 Issues 
* Code issue 126 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10897)

**Comment by sonaligoyal-harness** [comment]
trigger cypress

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/helper.ts [hunk 1]
```diff
@@ -189,3 +189,10 @@ export const updateMap = (alreadySelectedFields: JiraFieldNG[]): Record<string,
 export const isRuntimeOrExpressionType = (fieldType: MultiTypeInputType): boolean => {
   return fieldType === MultiTypeInputType.EXPRESSION || isMultiTypeRuntime(fieldType)
 }
+
+export const setUserValuesOptions = (userOptions: JiraUserData[]): MultiSelectOption[] => {
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/helper.ts:193]:
```suggestion
export const getOptionsFromJiraUsers = (userOptions: JiraUserData[]): MultiSelectOption[] => {
```

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/__tests__/JiraCreate.test.tsx [hunk 2]
```diff
@@ -307,6 +311,32 @@ describe('Jira Create tests', () => {
...
+                typeStr: '',
+                type: 'user'
+              },
+              required: false
+            }}
+            expressions={[]}
+          />
+        </JiraFieldsRenderer>
+      </TestWrapper>
+    )
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/__tests__/JiraCreate.test.tsx:336]: You can add more assertions to check if the users are being listed and if they can be selected.

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/JiraUserMultiTypeInput.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      userQuery: searchTerm,
+      connectorIdentifier: props.connectorRef,
+      offset: ''
+    },
+    debounce: 500
+  })
+  const handleQueryChange = (query: string) => {
+    /* istanbul ignore next */
+    return query ? setSearchTerm(query) : searchTerm
+  }
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/JiraUserMultiTypeInput.tsx:52]: If the query is an empty string `''`, updating the `searchTerm` to `''` could be useful as that will get more options from the API.

**Review by Eric761 — APPROVED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/__tests__/JiraCreate.test.tsx [hunk 4]
```diff
@@ -307,6 +311,37 @@ describe('Jira Create tests', () => {
...
+    expect(container).toMatchSnapshot()
+    await waitFor(() => {
+      fireEvent.change(container.querySelector('input[name="spec.selectedOptionalFields[0].value"]')!, {
+        target: { value: 'abhinav.singh3@harness.io' }
+      })
+    })
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/__tests__/JiraCreate.test.tsx:342]: `fireEvent.change` should not be inside `waitFor`. Add an assertion after the event to check the value.

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/JiraUserMultiTypeInput.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      placeholder={/* istanbul ignore next */ fetchUsers ? getString('loading') : selectedField.label}
+      disabled={isApprovalStepFieldDisabled(props.readonly) || fetchUsers}
+      className={cx(css.multiSelect, css.md)}
+      multiTypeInputProps={{
+        expressions,
+        selectProps: {
+          items: getUserValuesOptions(defaultTo(userData?.data, [])),
+          onQueryChange: handleQueryChange,
+          resetOnSelect: false
+        }
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/JiraUserMultiTypeInput.tsx:72]: Could you verify if the select retains the selected option when clicking outside to close the options?

Comment (sonaligoyal-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/JiraUserMultiTypeInput.tsx:72]: yes, it does

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**